src/main/java/com/coveo/nashorn_modules/Module.java

```diff
@@ -89,17 +110,20 @@ public Object require(String module) throws ScriptException {
     assert found != null;
     children.add(found.module);
 
+    //remove successful resolves from the refCache
```

Comment:
Also - if an exception is thrown the cache will be left with invalid values. You should add some handling for this to ensure that it's always cleaned up.